db/view/view_update_generator.cc

```diff
@@ -33,13 +35,23 @@ future<> view_update_generator::start() {
             }
             while (!_sstables_with_tables.empty()) {
                 auto& entry = _sstables_with_tables.front();
+              try {
                 schema_ptr s = entry.t->schema();
                 flat_mutation_reader staging_sstable_reader = entry.sst->read_rows_flat(s);
                 auto result = staging_sstable_reader.consume_in_thread(view_updating_consumer(s, _proxy, entry.sst, _as), db::no_timeout);
                 if (result == stop_iteration::yes) {
                     break;
                 }
+              } catch (...) {
+                vug_logger.warn("Processing {} failed: {}. Will retry...", entry.sst->get_filename(), std::current_exception());
+                break;
+              }
+              try {
                 entry.t->move_sstable_from_staging_in_thread(entry.sst);
+              } catch(...) {
+                // Move from staging will be retried upon restart.
+                vug_logger.warn("Moving {} from staging failed: {}. Ignoring...", entry.sst->get_filename(), std::current_exception());
+              }
```

Comment:
Now we will execute `pop_front()` and `signal()` below even if processing this sstable failed, which means this will be retried on restart.

I wonder if it would be better to just leave the entry and the semaphore, and just do a `continue` in the case of an error, so that the loop retries on the next iteration. In this case it would be best to use `semaphore_units` and add it to the `_sstables_with_tables`, letting `pop_front()` automatically signal the semaphore, instead of doing it manually.

BTW by "not in this patch" below I meant not in this PR. The rest of this PR looks good, I think this unrelated fix could be fixed in a separate one, allowing this PR to go in.

Reply:
@denesb, thanks for your review.

> Now we will execute pop_front() and signal() below even if processing this sstable failed, which means this will be retried on restart.

Actually, if processing stage throws, we break out of the loop so the entry will remain in `_sstables_with_tables` and will be retried on the next iteration.
Only moving out of the staging sub-directory is ignored and will be retried on restart.

Reply:
Ah, I see.